modules/govuk/manifests/apps/govuk_delivery.pp

```diff
@@ -43,66 +43,17 @@
 # [*enable_procfile_worker*]
 #   Run the celery worker defined in the procfile
 class govuk::apps::govuk_delivery(
-  $redis_host,
-  $redis_port,
-  $mongodb_hosts,
-  $mongodb_database,
-  $list_title_format,
-  $govdelivery_username = undef,
-  $govdelivery_password = undef,
-  $govdelivery_account_code = undef,
-  $govdelivery_hostname = undef,
-  $govdelivery_signup_form = undef,
   $port = '3042',
-  $enable_procfile_worker = true,
 ) {
   $app_name = 'govuk-delivery'
   include govuk_python
 
   govuk::app { $app_name:
+    ensure             => 'absent',
     app_type           => 'procfile',
     port               => $port,
     vhost_ssl_only     => true,
     health_check_path  => '/_status',
     log_format_is_json => true,
   }
-
-  govuk::procfile::worker { 'govuk-delivery':
-    enable_service => $enable_procfile_worker,
-  }
-
-  Govuk::App::Envvar {
```

Comment:
I'm not sure if we need to set these resources to `ensure => absent` to make sure they get removed (rather than stop managing the resource by removing the code), and then remove the entire class after applying puppet with the `ensure => absent` setting. That's what was done in https://github.com/alphagov/govuk-puppet/pull/6425/files and https://github.com/alphagov/govuk-puppet/pull/6430/files

Maybe @surminus can help to clarify? or the people involved in removing need-api

Reply:
Thanks @afda16 I've updated this in 716526ee0